src/gas/gas-util.test.ts

```diff
@@ -155,72 +227,741 @@ describe('gas utils', () => {
     });
   });
 
-  describe('normalizeGWEIDecimalNumbers', () => {
-    it('should convert a whole number to WEI', () => {
-      expect(normalizeGWEIDecimalNumbers(1)).toBe('1');
-      expect(normalizeGWEIDecimalNumbers(123)).toBe('123');
-      expect(normalizeGWEIDecimalNumbers(101)).toBe('101');
-      expect(normalizeGWEIDecimalNumbers(1234)).toBe('1234');
-      expect(normalizeGWEIDecimalNumbers(1000)).toBe('1000');
-    });
+  describe('fetchEthGasPriceEstimate', () => {
+    it('should fetch the current gas price and returns it as GWEI within an object', async () => {
+      const ethQuery = {
+        gasPrice(callback: any) {
+          callback(null, toHex('150340000000'));
+        },
+      };
 
-    it('should convert a number with a decimal part to WEI', () => {
-      expect(normalizeGWEIDecimalNumbers(1.1)).toBe('1.1');
-      expect(normalizeGWEIDecimalNumbers(123.01)).toBe('123.01');
-      expect(normalizeGWEIDecimalNumbers(101.001)).toBe('101.001');
-      expect(normalizeGWEIDecimalNumbers(100.001)).toBe('100.001');
-      expect(normalizeGWEIDecimalNumbers(1234.567)).toBe('1234.567');
-    });
+      const estimate = await fetchEthGasPriceEstimate(ethQuery);
 
-    it('should convert a number < 1 to WEI', () => {
-      expect(normalizeGWEIDecimalNumbers(0.1)).toBe('0.1');
-      expect(normalizeGWEIDecimalNumbers(0.01)).toBe('0.01');
-      expect(normalizeGWEIDecimalNumbers(0.001)).toBe('0.001');
-      expect(normalizeGWEIDecimalNumbers(0.567)).toBe('0.567');
+      expect(estimate).toStrictEqual({ gasPrice: '150.34' });
     });
+  });
 
-    it('should round to whole WEI numbers', () => {
-      expect(normalizeGWEIDecimalNumbers(0.1001)).toBe('0.1001');
-      expect(normalizeGWEIDecimalNumbers(0.0109)).toBe('0.0109');
-      expect(normalizeGWEIDecimalNumbers(0.0014)).toBe('0.0014');
-      expect(normalizeGWEIDecimalNumbers(0.5676)).toBe('0.5676');
-    });
+  describe('calculateTimeEstimate', () => {
+    const buildGasFeeEstimates = ({
+      estimatedBaseFee,
+      low = {},
+      medium = {},
+      high = {},
+    }: {
```

Comment:
Could we use the `GasFeeEstimates` type directly here rather than re-declaring it?